lib/src/view/menu.dart

```diff
@@ -55,7 +55,8 @@ UiFactory<MenuProps> ConnectedMenu = connect<AppState, MenuProps>(
     ..display_major_tick_widths_all_helices = state.ui_state.display_major_tick_widths_all_helices
     ..invert_y_axis = state.ui_state.invert_y_axis
     ..show_helix_circles_main_view = state.ui_state.show_helix_circles_main_view
-    ..warn_on_exit_if_unsaved = state.ui_state.warn_on_exit_if_unsaved),
+    ..warn_on_exit_if_unsaved = state.ui_state.warn_on_exit_if_unsaved)
```

Comment:
This looks wrong. The parentheses should be after the last props variable is set. Now that `show_grid_coordinates_side_view` is the last, the last two lines should be 

```dart
    ..warn_on_exit_if_unsaved = state.ui_state.warn_on_exit_if_unsaved
    ..show_grid_coordinates_side_view = state.ui_state.show_grid_coordinates_side_view),
```

I'm surprised this even compiles. You've tested this and it works?

Reply:
Not sure, when I ran both the unit tests and the live server everything looked like it was working. I'll change this and the rest of the issues and re-update.

Reply:
Yes, I checked out this commit and was able to run it. I think it works because the parentheses surround the `Menu()` constructor and subsequent field assignments, and the value of the expression in the parentheses is `Menu`, so it's possible to assign to its fields outside of the parentheses. I think maybe the parentheses are unnecessary in this case.

But outside of a `connect` function, the typical way to make a React component looks like this:

```dart
return (Dom.rect()
          ..x = background_x
          ..y = background_y
          ..width = background_width
          ..height = background_height
          ..className = 'insertion-background'
          ..onClick = ((_) => change_insertion_length())
          ..key = 'rect')();
```

If I didn't set any of those properties, and removed the parentheses around `Dom.rect()`, it would look like this:

```dart
return Dom.rect()();
```

`Dom.rect()` returns an object of type `UiProps` (in this case `SvgProps`, but for custom components, it's the type of the `Props` mixin, e.g., for `MenuBoolean` it's of type `MenuBooleanProps`, whose fields are listed in `MenuBooleanPropsMixin`). That's the object that you set the properties on. Once you're done, that object is callable like a function, so you call it by putting parentheses again. Either it takes no argument, or the arguments specified the children in the View tree (if you look around, you can see some `Dom.g` and `Dom.div` elements always have arguments in the second set of parentheses). The return value of *that* call is the actual `ReactElement` object.

If you tried without the parentheses when specifying one or more props, then it wouldn't work:

```dart
//XXX: compilation error, don't do this
return Dom.rect()
          ..x = background_x
          ..y = background_y
          ..width = background_width
          ..height = background_height
          ..className = 'insertion-background'
          ..onClick = ((_) => change_insertion_length())
          ..key = 'rect'();
```

I think the difference is that in `connect` and `mapStateToProps`, you don't do the second set of parentheses, because `mapStateToProps` is a function that returns a `UiProps` object, not a `ReactElement`. But I just got in the habit of putting parentheses around the whole thing anyway.